model/host_manipulation.go

```diff
@@ -20,42 +21,74 @@ func UpdateStaticHosts() error {
 	if err != nil {
 		return err
 	}
-	activeStaticHosts := make([]string, 0)
-
+	activeHosts := []string{}
+	catcher := grip.NewBasicCatcher()
 	for _, d := range distros {
-		settings := &cloud.StaticSettings{}
-		err = mapstructure.Decode(d.ProviderSettings, settings)
+		hosts, err := doStatcHostUpdate(d)
 		if err != nil {
-			return errors.Errorf("invalid static settings for '%v'", d.Id)
+			catcher.Add(err)
+			continue
 		}
-		for _, h := range settings.Hosts {
-			hostInfo, err := util.ParseSSHInfo(h.Name)
-			if err != nil {
-				return err
-			}
-			user := hostInfo.User
-			if user == "" {
-				user = d.User
-			}
-			staticHost := host.Host{
-				Id:           h.Name,
-				User:         user,
-				Host:         h.Name,
-				Distro:       d,
-				CreationTime: time.Now(),
-				Provider:     evergreen.HostTypeStatic,
-				StartedBy:    evergreen.User,
-				Status:       evergreen.HostRunning,
-				Provisioned:  true,
-			}
-
-			// upsert the host
-			_, err = staticHost.Upsert()
-			if err != nil {
-				return err
-			}
-			activeStaticHosts = append(activeStaticHosts, h.Name)
+		activeHosts = append(activeHosts, hosts...)
+
+	}
+	if catcher.HasErrors() {
+		return catcher.Resolve()
+	}
+
+	return host.MarkInactiveStaticHosts(activeHosts)
+}
+
+func UpdateStaticDistro(name string) error {
```

Comment:
this function is unused

Reply:
soon! sorry for splitting this up less clearly than I could